src/screens/Settings/hooks.ts

```diff
@@ -36,3 +40,27 @@ export const useToggleAnalytics = () => {
 
   return { analyticsEnabled, toggleAnalytics };
 };
+
+/**
+ * Hook that provides a function to enable or disable the
+ * prompt for unlocking the application and the current status.
+ */
+export const useToggleAppLock = () => {
+  const { t } = useTranslation('settings');
+  const navigation = useNavigation<StackNavigationProp<RootNavigatorParamList>>();
+  const appLockEnabled = useSetting('autoAppLock');
+  const setAppLockEnabled = useSetSetting('autoAppLock');
+
+  const toggleAppLock = React.useCallback(() => {
+    navigation.navigate(ROUTES.SETTINGS_SWITCH_SCREEN, {
+      title: t('disable app lock'),
+      description: t('disable app lock message'),
+      intialValue: !appLockEnabled,
+      toggleSetting: () => {
+        setAppLockEnabled((currentValue) => !currentValue);
+      },
+    });
+  }, [appLockEnabled, navigation, setAppLockEnabled, t]);
+
+  return { toggleAppLock };
```

Comment:
Why we don't simply return the function instead of an object containing a single field?